flink/src/main/java/org/apache/iceberg/flink/FlinkFilters.java

```diff
@@ -0,0 +1,267 @@
+  }
+
+  private static Optional<Expression> convertLogicExpression(BiFunction<Expression, Expression, Expression> function,
+                                                             CallExpression call) {
+    List<ResolvedExpression> args = call.getResolvedChildren();
+    Optional<Expression> left = convert(args.get(0));
+    Optional<Expression> right = convert(args.get(1));
+    if (left.isPresent() && right.isPresent()) {
+      return Optional.of(function.apply(left.get(), right.get()));
+    }
+
+    return Optional.empty();
+  }
+
+  private static Optional<Expression> convertComparisonExpression(
+      BiFunction<String, Object, Expression> function, BiFunction<String, Object, Expression> reversedFunction,
+      CallExpression call) {
+    Tuple2<String, Object> tuple2 = convertBinaryExpress(call);
+    if (tuple2 != null) {
+      if (literalOnRight(call.getResolvedChildren())) {
+        return Optional.of(function.apply(tuple2.f0, tuple2.f1));
+      } else {
+        return Optional.of(reversedFunction.apply(tuple2.f0, tuple2.f1));
+      }
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  private static Optional<String> toReference(org.apache.flink.table.expressions.Expression expression) {
+    return expression instanceof FieldReferenceExpression ?
+        Optional.of(((FieldReferenceExpression) expression).getName()) :
+        Optional.empty();
+  }
+
+  private static Optional<Object> toLiteral(org.apache.flink.table.expressions.Expression expression) {
+    // Not support null literal
+    return expression instanceof ValueLiteralExpression ?
+        convertLiteral((ValueLiteralExpression) expression) :
+        Optional.empty();
+  }
+
+  private static Optional<Expression> handleNaN(BiFunction<String, Object, Expression> function,
+                                                Function<String, Expression> functionNaN,
+                                                CallExpression call) {
+    Tuple2<String, Object> tuple2 = convertBinaryExpress(call);
+    if (tuple2 == null) {
+      return Optional.empty();
+    }
+
+    String name = tuple2.f0;
+    Object value = tuple2.f1;
+
+    if (NaNUtil.isNaN(value)) {
+      return Optional.of(functionNaN.apply(name));
+    } else {
+      return Optional.of(function.apply(name, value));
+    }
+  }
+
+  private static Optional<Object> convertLiteral(ValueLiteralExpression expression) {
+    Optional<?> value = expression.getValueAs(expression.getOutputDataType().getLogicalType().getDefaultConversion());
+    return value.map(o -> {
+      if (o instanceof LocalDateTime) {
+        return DateTimeUtil.microsFromTimestamp((LocalDateTime) o);
+      } else if (o instanceof Instant) {
+        return DateTimeUtil.microsFromInstant((Instant) o);
+      } else if (o instanceof LocalTime) {
+        return DateTimeUtil.microsFromTime((LocalTime) o);
+      } else if (o instanceof LocalDate) {
+        return DateTimeUtil.daysFromDate((LocalDate) o);
```

Comment:
If the literal type of flink is BINARY or VARBINARY,   then its java data type is  `byte[]`,  while in iceberg we will use `ByteBuffer` to do the literal comparison.  So I think we need to convert it to ByteBuffer by `ByteBuffer.wrap((byte[])o)` .  Otherwise the iceberg literal comparing will throw a cast failure . 

Pls see the java data type for iceberg type  here: https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/types/Type.java#L30

By the way,  I think you unit tests here https://github.com/apache/iceberg/pull/1893/files#diff-5d18d1ff127d1dc70a9a15bbe941f2b6f9d28b3015924f601ac1f722914099dbR81 is incorrect.

Reply:
I  checkout the code ,I found that iceberg already do the convertion, ([here](https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/expressions/Literals.java#L77)) ,so I did not add the conversion here.

Reply:
OK,  that make sense.